test/GoalTest.php

```diff
@@ -0,0 +1,285 @@
+<?php
+
+namespace Acquia\LiftClient\Test;
+
+use Acquia\LiftClient\Entity\Goal;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Psr7\Response;
+
+class GoalTest extends TestBase
+{
+    public function testGoalAdd()
+    {
+        // Setup
+        $data = [
+          'id' => 'test-id',
+          'name' => 'test-name',
+          'description' => 'test-description',
+          'rule_ids' => [
+            'rule-id-1',
+          ],
+          'site_ids' => [
+            'site-id-1',
+          ],
+          'event_names' => [
+            'Click-Through',
+          ],
+          'global' => false,
+          'value' => 100,
+        ];
+        $response = new Response(200, [], json_encode($data));
+        $responses = [
+          $response,
+        ];
+
+        $client = $this->getClient($responses);
+
+        // Create a new Goal object.
+        $goal = new Goal();
+        $goal->setName('test-name');
+        $goal->setId('test-id');
+        $goal->setDescription('test-description');
+        $goal->setRuleIds(array('rule-id-1'));
+        $goal->setSiteIds(array('site-id-1'));
+        $goal->setEventNames(array('Click-Through'));
+
+        // Get Slot Manager
+        $manager = $client->getGoalManager();
+        $response = $manager->add($goal);
+
+        // Check if the identifier is equal.
+        $this->assertEquals($response->getId(), $goal->getId());
+        // Check if the description is equal.
+        $this->assertEquals(
+          $response->getDescription(),
+          $goal->getDescription()
+        );
+        // Check if the label is equal.
+        $this->assertEquals($response->getName(), $goal->getName());
+
+        // Check if the description is equal.
+        $this->assertEquals($response->getDescription(), $goal->getDescription());
+
+        // Check if the rule_ids is equal.
+        $this->assertEquals($response->getRuleIds(), $goal->getRuleIds());
+
+        // Check if the site_ids is equal.
+        $this->assertEquals($response->getSiteIds(), $goal->getSiteIds());
+
+        // Check if the site_ids is equal.
+        $this->assertEquals($response->getEventNames(), $goal->getEventNames());
+
+        // Check if the global was set correctly.
+        $this->assertEquals($response->getGlobal(), $goal->getGlobal());
```

Comment:
I was taught once that it is better to use a plain value (like a plain string) in the "expected" field. No biggie either way though.